GVFS/GVFS.Common/NamedPipes/NamedPipeMessages.cs

```diff
@@ -137,6 +138,46 @@ public Message CreateMessage()
             }
         }
 
+        public static class PostIndexChanged
+        {
+            public const string NotificationRequest = "PICN";
+            public const string SuccessResult = "S";
+            public const string FailureResult = "F";
+
+            public class Request
+            {
+                public Request(Message message)
+                {
+                    if (message.Body.Length != 2)
+                    {
+                        throw new InvalidOperationException($"Invalid PostIndexChanged message. Expected 2 characters, got: {message.Body.Length} from message: '{message.Body}'");
+                    }
+
+                    this.WorkingDirectoryUpdated = message.Body[0] == '1';
+                    this.SkipWorktreeBitsUpdated = message.Body[1] == '1';
+                }
+
+                public bool WorkingDirectoryUpdated { get; }
+
+                public bool SkipWorktreeBitsUpdated { get; }

```

Comment:
This sounds like a request - "Skip the worktree bits update"

I would reword both of these: `UpdatedWorkingDirectory` and `UpdatedSkipWorktreeBits`